src/pages/index.js

```diff
@@ -106,45 +175,89 @@ function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener('keyup', handleEscape);
 }
+
 function closeModal(modal) {
   modal.classList.remove("modal_opened");
   document.removeEventListener('keyup', handleEscape);
 }
 
+
+
 function handleEditFormSubmit(event) {
   event.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+
+  const submitButton = event.submitter;
+  submitButton.textContent = "Saving...";
+
+  api
+    .editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((_data) => {
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+}
+
+ function handleAddCardSubmit(event) {
+   event.preventDefault();
+
+   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+   const submitButton = event.submitter;
+   submitButton.textContent = "Saving...";
+
+  api
+    .postCards(inputValues)
+    .then((res) => {
+      const cardElement = getCardElement(res);
+      cardsList.prepend(cardElement);
+      event.target.reset();
+      closeModal(cardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
 }
-function handleAddCardSubmit(event) {
+
+function handleAvatarSubmit(event) {
   event.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  event.target.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+
+  const submitButton = event.submitter;
+  submitButton.textContent = "Saving...";
+
+  api
+    .editAvatarInfo(avatarinput.value)
+    .then((res) => {
+      document.querySelector(".profile__avatar").src = res.avatar;
```

Comment:
`document.querySelector(".profile__avatar")`

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine